### ISSUE TITLE
bug(e18e/prefer-array-to-sorted): false postive with non-arrays

### ISSUE THREAD

**Issue by WarningImHack3r** [e18e/eslint-plugin#54]
The rule `e18e/prefer-array-to-sorted` is indeed a good one, but it doesn't check whether the input is actually an array. For example:
```ts
const s = new Set(["a", "b", "c"]); // dummy array
const result = [...s].sort(); // ← triggers the rule incorrectly, `s` is not an array we can use `toSorted` on!
```

It also falls apart with `Iterator`s and other constructs that we have to use spread on to use array-specific functions.

It's also "hard" to bypass, because even if we can replace this code with `[...s].toSorted()` to mute the issue, `toSorted` creates yet another copy, which is the opposite of what this plugin exists for ^^
As a result, I need to disable it for this specific line waiting for a fix.
